pyiceberg/table/metadata.py

```diff
@@ -313,6 +315,34 @@ def construct_partition_specs(cls, data: Dict[str, Any]) -> Dict[str, Any]:
 
         return data
 
+    @model_validator(mode="before")
+    def construct_v1_spec_from_v2_fields(cls, data: Dict[str, Any]) -> Dict[str, Any]:
+        specs_field = "partition_specs"
+        default_spec_id_field = "default_spec_id"
+        if specs_field in data and default_spec_id_field in data:
+            found_spec = next((spec for spec in data[specs_field] if spec.spec_id == data[default_spec_id_field]), None)
+            if found_spec is not None:
+                spec_dict = found_spec.model_dump()
+                spec_dict['fields'] = list(spec_dict['fields'])
```

Comment:
If I don't do this conversion from the tuple to a list what ends up happening is the read metadata from the JSON is a list but then the actual model in spec_dict['fields'] is a tuple. Since partition_spec is deprecated anyways and is a List[Dict[str, any]] I think we have the flexibility just to make the fields a list.